app/views/administrate/application/_collection_header_actions.html.erb

```diff
@@ -1,4 +1,8 @@
-<% [valid_action?(:edit, collection_presenter.resource_name),
-    valid_action?(:destroy, collection_presenter.resource_name)].count(true).times do %>
+<% [
+    page.try(:resource).present? && valid_action?(:edit, collection_presenter.resource_name, page.resource) ||
+    valid_action?(:edit, collection_presenter.resource_name),
+    page.try(:resource).present? && valid_action?(:destroy, collection_presenter.resource_name, page.resource) ||
+    valid_action?(:destroy, collection_presenter.resource_name)
```

Comment:
I think it's possible to simplify these two:

```suggestion
    valid_action?(:edit, collection_presenter.resource_name, page.try(:resource).presence),
    valid_action?(:destroy, collection_presenter.resource_name, page.try(:resource).presence)
```

Reply:
Well spotted, thanks!